File: 2015/template.py
"""

Advent of Code 2015, day
Solution by Geir Arne Hjelle, 2016-12-
"""
import pathlib
import sys

debug = print if "--debug" in sys.argv else lambda *_: None


def main(args):
    """Solve the problem for all file paths"""
    for file_path in [pathlib.Path(p) for p in args if not p.startswith("-")]:
        solve(file_path)


def solve(file_path):
    """Solve the problem for one file path"""
    print(f"\n{file_path}:")
    text = file_path.read_text()
    print(text)


if __name__ == "__main__":
    main(sys.argv[1:])

File: 2015/template.jl
#
#
# Advent of Code 2015, day
# Solution by Geir Arne Hjelle, 2020-12-

using Pipe

# Solve the problem for one file
function solve(filename)
    println("\n$(filename)")

    # Read from file
    input = open(filename) do fid
        fid |> readlines
    end

    # Part 1
    input .|> println
end


# Solve the problem for each file
ARGS .|> solve
